Sources/APPX.cpp

```diff
@@ -102,18 +104,40 @@ namespace appx {
             for (const auto &fileNamePair : fileNames) {
                 const std::string &archiveName = fileNamePair.first;
                 const std::string &fileName = fileNamePair.second;
+
+                const std::string suffix = "AppxBundleManifest.xml";
+                if (bundle &&
+                        suffix.size() < archiveName.size() &&
+                        std::equal(suffix.rbegin(), suffix.rend(), archiveName.rbegin())) {
+                    appxBundleManifest = fileNamePair;
+                    continue;
+                }
+
                 zipFileEntries.emplace_back(
                     WriteZIPFileEntry(sink, zipOffsetSink.Offset(), fileName,
                                       archiveName, compressionLevel));
             }
 
+            if (bundle) {
+                ZIPFileEntry appxBundleManifestEntry = WriteAppxBundleManifestZIPFileEntry(
+                    sink,
+                    zipOffsetSink.Offset(),
+                    appxBundleManifest.second,
+                    appxBundleManifest.first,
+                    compressionLevel,
+                    zipFileEntries);
+                zipFileEntries.emplace_back(std::move(appxBundleManifestEntry));
```

Comment:
Nit: `appxBundleManifestEntry` can be inlined (and then the `std::move` call dropped).